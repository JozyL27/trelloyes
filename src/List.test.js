import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import List from './List';



describe('<List />', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');

        // Pass required props in the usual way
        ReactDOM.render(
            <List 
            header="Joe"
            cards={[{id: 89, title: "shoe", content: "nike"}, {id: 69, title: "shoes", content: "adidas"}]}
            />, div
        );

        ReactDOM.unmountComponentAtNode(div);
    });
});


describe('<List />', () => {
    it('renders ui as expected', () => {
        const tree = renderer.create(
            <List 
            header="joe"
            cards={[{id: 89, title: "shoe", content: "nike"}, {id: 69, title: "shoes", content: "adidas"}]}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
})

// header = h2 = string
// cards = array = array of objects