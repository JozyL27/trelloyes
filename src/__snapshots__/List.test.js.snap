// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders ui as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      joe
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        shoe
      </h3>
      <p>
        nike
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        shoes
      </h3>
      <p>
        adidas
      </p>
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
